java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/OsmAndMapsService.java

```diff
@@ -880,25 +870,35 @@ private synchronized List<RouteSegmentResult> approximateInternal(RoutingContext
 		putResultProps(ctx, route, props);
 		return route;
 	}
-	
+
 
 	private static class RouteParameters {
 		String routeProfile;
 		Map<String, String> routeParams = new LinkedHashMap<String, String>();
 		public RouteParameters(String p) {
 			this.routeProfile = p;
+			final String CAR = "car";
+			Set<String> complexModeProfiles = new HashSet<>();
```

Comment:
Doesn't look nice ```RoutingConfiguration.getDefault().getAllRouters()...``` Too many things goes on in constructor

Reply:
fixed (removed because rm became nullable)